docs/ccdproc/image_management.rst

```diff
@@ -0,0 +1,141 @@
+
+    >>> ic1.summary.colnames
+    ['imagetyp', 'object', 'filter']
+    >>> ic_all.summary.colnames
+    # long list of keyword names omitted
+
+Selecting files
+---------------
+
+Selecting the files that match a set of criteria, for example all images in
+the I band with exposure time less than 60 seconds you could do::
+
+    >>> matches = (ic1.summary['filter'] == 'I' & ic1.summary['exposure'] < 60)
+    >>> my_files = summary['file'][matches]
+
+The column ``file`` is added automatically when the image collection is created.
+
+For more simple selection, when you just want files whose keywords exactly
+match particular values, say all 'I' band images with exposure time of 30
+seconds, there is a convenience method ``.files_filtered``::
+
+    >>> my_files = ic1.files_filtered(filter='I', exposure=30)
+
+The optional arguments to ``files_filtered`` are used to filter the list of
+files.
+
+Iterating over hdus, headers or data
+------------------------------------
+
+Three methods are provided for iterating over the images in the collection,
+optionally filtered by keyword values.
+
+For example, to iterate over all of the I band light images with exposure of
+30 seconds, performing some basic operation on the data (very contrived
+example)::
+
+    >>> for hdu in ic1.hdus(imagetyp='LiGhT', filter='I', exposure=30):
+    ...     hdu.header['exposure']
+    ...     new_data = hdu.data - hdu.data.mean()
+
+Note that the names of the arguments to ``hdus`` here are the names of FITS
+keywords in the collection and the values are the values of those keywords you
+want to select. Note also that string comparisons are not case sensitive.
+
+The other iterators are ``headers`` and ``data``.
+
+All of them have the option to also provide the file name in addition to the
+hdu (or header or data)::
+
+    >>> for hdu, fname in ic1.hdus(return_fname=True,
+    ...                            imagetyp='LiGhT', filter='I', exposure=30):
+    ...    hdu.header['meansub'] = True
+    ...    hdu.data = hdu.data - hdu.data.mean()
+    ...    hdu.writeto(fname + '.new')
+
+That last use case, doing something to several files and wanting to same them
+somewhere afterwards, is common enough that the iterators provide arguments to
+automate it.
+
+Automatic saving from the iterators
+-----------------------------------
+
+There are three ways of triggering automatic saving.
+
+1. One is with the argument ``save_with_name``; it adds the value of the
+argument to the file name between the original base name and extension. The
+example below has (almsot) the same effect of the example above, subtracting
+the mean from each image and saving to a new file::
+
+    >>> for hdu in ic1.hdus(save_with_name='_new',
+    ...                     imagetyp='LiGhT', filter='I', exposure=30):
+    ...    hdu.header['meansub'] = True
+    ...    hdu.data = hdu.data - hdu.data.mean()
+
+   It saves, in the ``location`` of the image collection, a new FITS file with
```

Comment:
This ends up a little mangled on build.  Need to adjust the indentiion starting with 'It'.